Core/src/Exception/ServiceException.php

```diff
@@ -81,4 +86,66 @@ public function getMetadata()
     {
         return $this->metadata;
     }
+
+    /**
+     * Returns the metadata from the ErrorInfo part of the exception
+     *
+     * @return array
+     */
+    public function getErrorInfoMetadata()
+    {
+        // For response originated from the GAPIC layer, the current exception would have
+        // an ApiException within itself
+        if ($this->getServiceException() instanceof ApiException) {
+            return $this->getServiceException()->getErrorInfoMetadata();
+        }
+
+        $errorInfo = $this->getErrorInfoFromRestException();
```

Comment:
WDYT about caching the value returned from here in memory so that if someone calls both `getErrorInfoMetadata` and `getReason` the loop isn't cycled twice?